docs/ai-instructions/dependencies-component-instances.md

```diff
@@ -0,0 +1,102 @@
+
+### Automated Fix Process
+
+#### Step 1: Check Component Files
+
+For each cartridge with `*.component` files:
+
+1. Scan all `.component` files in `src/main/resources/resources/{cartridge}/components/`
+2. Look for `<fulfill requirement="app" with="intershop.{ApplicationType}" />` patterns
+3. List all ApplicationType instances found in the cartridge
+
+#### Step 2: Validate Component Placement
+
+- **Correct**: Components match the cartridge's designated ApplicationType from the table above
+- **Incorrect**: Components belong to different ApplicationTypes than the cartridge's designation (see "Category" in the table above)
+- **Action**: Move misplaced components to appropriate cartridge or `as_` cartridge
+
+#### Step 3: Fix Misplaced Components
+
+For components that don't belong in their current cartridge:
+
+1. **Target exists**: Move component to the correct cartridge (e.g., `intershop.B2CBackoffice` → `sld_ch_consumer_plugin_<project>`)
+2. **Target doesn't exist**: Move component to appropriate `as_` cartridge (e.g., `as_backoffice_<project>`)
+3. **Update dependencies**: Ensure `as_` cartridge has `cartridgeRuntime` dependencies to all referenced cartridges
+
+#### Step 4: Dependency Validation
+
+For each cartridge that provides ApplicationType instances:
+
+1. **Check `build.gradle.kts`**: Must have `cartridge()` or `cartridgeRuntime()` dependency matching the table above
+2. **Dependency Type Rules**:
+   - Use `cartridge()` if the cartridge needs compile-time access (like `implementation`)
+   - Use `cartridgeRuntime()` only if runtime access is sufficient (like `runtimeOnly`)
+   - **Important**: If `cartridge()` dependency exists, `cartridgeRuntime()` is redundant and should be removed
+3. **Missing dependency**: Add the required external dependency (e.g., `cartridgeRuntime("com.intershop.business:sld_enterprise_app")`) if component files/content is moved.
+4. **Extra dependencies**: Only remove if they're not project-specific or referenced in component files
+
+### Examples
+
+**Good**: `sld_enterprise_app_<project>` only contains components for `intershop.EnterpriseBackoffice`
+**Bad**: `sld_enterprise_app_<project>` contains components for `intershop.B2CBackoffice` and `intershop.PartnerBackoffice`
+**Fix**: Move `B2CBackoffice` and `PartnerBackoffice` components to `as_backoffice_<project>` or `as_<project>` if exists.
+
+### Validation Commands
+
+- Find all cartridges with component files: `find . -name "*.component" -path "*/components/*"`
+- Check ApplicationType references: `grep -r "intershop\." --include="*.component" */components/`
+- Verify dependencies: Check each cartridge's `build.gradle.kts` for required `cartridgeRuntime` entries
+
+### AI Task - Instruction
+
+- validate my cartridge dependencies and component files. see component-instances.md. Are there cartridges, with missing dependencies.
```

Comment:
is the file name here correct? the new instructions file has the prefix `dependencies-`.
```suggestion
- validate my cartridge dependencies and component files. see `dependencies-component-instances.md`. Are there cartridges, with missing dependencies.
```